### PR TITLE
bandominedoni same note fix

### PR DESCRIPTION
**Pull request by 3araht**
## Description

 An issue with multiple same notes played at the same time is fixed.
(This version requires https://github.com/qmk/qmk_firmware/pull/20179 to be implemented to use an encoder correctly.)

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by 3araht** [comment]
> Not on the computer right now, so cant really check to confirm it... Changes to move information into data driven (read: info.json) are most likely done on the develop branch already, thus would lead to conflicts when develop gets merged on master, changes to the keyboard's code should target develop
> 
> Also, there seems to be quite some duplication (eg the custom keycodes across several keymaps), which should be moved to keyboard-level (and use QK_KB instead of SAFE_RANGE)

I made a branch from a develop branch and applied all the modifications done here to avoid conflicts.
I'd like to send a pull request to a develop branch, but I mistakenly pull-requested the whole develop branch to master.
Maybe I should wait for the next breaking change so that I can send a pull request as usual.

**Comment by elpekenin** [comment]
On my computer now, so i can take a closer look:
1. Indeed, the changes to move stuff into the JSON file are already done in https://github.com/qmk/qmk_firmware/blob/develop/keyboards/bandominedoni/info.json#L12-L13, so you should not make such thing on a PR to `master`
2. As i said a few hours ago, you have tons of duplication across files (defining a lot of custom keycodes and their behaviour, on each of the `keymap.c` files), that should be moved to `bandominedoni.c` using `process_record_kb` (remember to call the `user` variant from there): 
```c
bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
    if (!process_record_user(keycode, record) {
        return false;
    }

    // your code
}
```

**Comment by 3araht** [comment]
> On my computer now, so i could take a closer look:
> 
> 1. Indeed, the changes to move stuff into the JSON file are already done in https://github.com/qmk/qmk_firmware/blob/develop/keyboards/bandominedoni/info.json, so you should not make such thing on a PR to `master`
> 2. As i said a few hours ago, you have tons of duplication across files (defining a lot of custom keycodes and their behaviour, on each of the `keymap.c` files), that should be moved to `bandominedoni.c` using `process_record_kb` (remember to call the `user` variant from there):
> 
> ```c
> bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
>     if (!process_record_user(keycode, record) {
>         return false;
>     }
> 
>     // your code
> }
> ```

Oh, I see. I moved some Macro definitions and functions to there but I didn't modify process_record_kb() yet.
I'll do that.

**Review by drashna — APPROVED** [review]

File: keyboards/bandominedoni/bandominedoni.h [hunk 1]
```diff
@@ -77,9 +77,141 @@
...
 
+//  define which MIDI ch to use.
+//  Note that (actual MIDI ch# - 1) -> 0 .. 15 is used for coding.
+//  ch1
+#define DEFAULT_SUB_CH_NUMBER  0
+//  ch2
+#define ALT_SUB_CH_NUMBER      1
+
+enum custom_keycodes {
+    VERSION = QK_KB,
```
Comment (zvecr) [inline keyboards/bandominedoni/bandominedoni.h:88]: You only get 64 keyboard level keycodes <https://github.com/qmk/qmk_firmware/blob/master/data/constants/keycodes/keycodes_0.0.2.hjson#L6>, and can count 75 which would be invalid.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/bandominedoni/bandominedoni.h (hunk 1 above)
Comment (3araht) [reply keyboards/bandominedoni/bandominedoni.h:88]: Uh, oh…
I can’t use QK_USER in the KB level, can I?

BYW, Thanks for changing the base branch to develop.

**Pull request closed without merging**